### PR TITLE
Bumping up the test coverage further

### PR DESCRIPTION
**Pull request by piyushghai**
1. Added test cases for codec.py
2. Added .coveragearc file to check the code coverage metrics only for the backend files. 

The command to run it is : (from the root project folder)

```py.test --cov-report  term-missing --cov=mms/ --ignore mms/tests/integration_tests/```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+branch = True
+omit = */__init__.py
+       mms/tests/*
+       mms/utils/model_server_error_codes.py
+       mms/utils/timeit_decorator.py
+       mms/storage.py
+       mms/metrics/system_metrics.py
+       mms/utils/mxnet/*
+       mms/examples/metric_push_example.py
+       mms/model_service/*
```
Comment (zachgk) [inline .coveragerc]: Missing newline at end of file.  https://stackoverflow.com/questions/16761227/how-to-make-intellij-idea-insert-a-new-line-at-every-end-of-file might help

File: .coveragerc [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
```
Comment (zachgk) [inline .coveragerc]: You may want the command to run the tests and check coverage in a README

Comment (piyushghai) [reply .coveragerc]: Good point. Done. 8dfa551a22b89525d9f9b66bc906219c6d9aed94

File: .coveragerc (hunk 1 above)
Comment (piyushghai) [reply .coveragerc]: Done. Thanks! 8dfa551a22b89525d9f9b66bc906219c6d9aed94

File: mms/tests/unit_tests/test_utils/test_codec_helpers.py [hunk 3]
```diff
@@ -16,3 +18,36 @@ def test_invalid_encode(self):
...
+            b64_mock = mocker.patch('base64.b64decode')
+            exception = binascii.Error('error')
+            b64_mock.side_effect = exception
+
+            with pytest.raises(Exception) as ex:
+                ModelWorkerCodecHelper.decode_msg('base64', self.encoded_msg)
+
+            assert isinstance(ex.value, MMSError)
+            assert ex.value.get_code() == err.DECODE_FAILED
+            assert ex.value.get_message() == "base64 decode error {}".format(repr(exception))
```
Comment (vrakesh) [inline mms/tests/unit_tests/test_utils/test_codec_helpers.py]: can't str() do this?

Comment (zachgk) [reply mms/tests/unit_tests/test_utils/test_codec_helpers.py]: They are slightly different.  For exceptions, str(Exception('msg')) == 'msg' and repr(Exception('msg')) == "Exception('msg')"

Comment (piyushghai) [reply mms/tests/unit_tests/test_utils/test_codec_helpers.py]: Here's the difference : 
str(exception) prints just the message part of the exception.

repr(exception) prints the exception message as well as the exception class.

**Pull request merged**